full overhaul of site styles and interactivity

hagercreative opened the pull request.

mhagercp committed:
full overhaul of site styles and interactivity

NekoCarreraDesigns reviewed (changes requested):
Hello Mark,

Disclaimer: don't take this review personally, you are not your code, this is constructive criticism. not a personal attack on you. You'd be surprised how often this happens. 

Let's start from the top the .gitignore you have needs this code in it
# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.

# dependencies
/node_modules
/.pnp
.pnp.js

# testing
/coverage

# production
/build

# misc
.DS_Store
.env.local
.env.development.local
.env.test.local
.env.production.local
config.env
jwt.env

npm-debug.log*
yarn-debug.log*
yarn-error.log

the original one should have never been deleted, you should have deleted the one you made. move this into the main eFormative folder.

background images might not be showing because they are in the background property instead of background-image property, this might be why they are not displaying the images from the imgur links, also make sure you are using the image link address, instead of the link address.  Where these can be found are index.css, Home.css.

Market.css - .search-button:hover hex decimal code has extra characters
.clear-button:hover has extra characters in hex decimal code too

Reviews.css hex decimal codes missing characters in the classes .review-header, .reviews-div 
 and .product-search-button:hover has extra hex decimal code characters

Sell.css .user-signup-redirect:hover hex decimal code has extra characters

Notes on frontend code, stylistically speaking it's well written.  Few things I would like to express to you. When someone uses single quotes originally you can keep the single quotes no need to change them to doubles since JavaScript recognizes both for strings. if it was C languages or Python that would be a good catch. Sticking with the same properties is also something to consider, I use rgb because its easy to write for me, hex decimal codes I have to look up every time. there is sites that convert hex decimals to rgb. making sure everything is similar is always good when collaborating because it keeps everything nice and organized.

I would like you to fix whatever happened to the node_modules in the server folder, Why did you rewrite these? Its very bad  practice to rewrite someone else's code without their permission especially node modules or npm packages that have been published. I specifically asked in the beginning that you stayed out of this folder, and should only go into it to install the modules, so you could run the files. Whatever happened in this folder it wants to delete the entire package-lock.json which would destroy the backend entirely. this has to be fixed before I can merge this branch

other than the stylistic differences, the code is well written and readable.

hagercreative commented:
Thanks for the feedback, I know how critiques can go from art school, so no
offense taken. I'm very new to this, so I appreciate any and all feedback
you have. I'll work on making all these changes soon, but I'll give some
initial responses that I have to a few of these here.

See https://help.github.com/articles/ignoring-files/ for more about
> ignoring files. dependencies
>
> /node_modules
> /.pnp
> .pnp.js
> testing
>
> /coverage
> production
>
> /build
> misc
>
> .DS_Store
> .env.local
> .env.development.local
> .env.test.local
> .env.production.local
> config.env
> jwt.env
>
> npm-debug.log*
> yarn-debug.log*
> yarn-error.log
>
> the original one should have never been deleted, you should have deleted
> the one you made. move this into the main eFormative folder.
>
*I didn't know I deleted one, it was an error if I had, thanks for the
correct gitignore.*

> background images might not be showing because they are in the background
> property instead of background-image property, this might be why they are
> not displaying the images from the imgur links, also make sure you are
> using the image link address, instead of the link address. Where these can
> be found are index.css, Home.css.
>
*I'll look into this, all the imgur links ARE displaying correctly for me,
it was the local ones that I tried to use as backgrounds that didn't. I'll
double check with background-image, but I was sure I tried that.*

> Market.css - .search-button:hover hex decimal code has extra characters
> .clear-button:hover has extra characters in hex decimal code too
>
*The extra characters are the alpha channel. They adjust the opacity of the
hex color. Since I'm primarily a designer, and I created all his brand
colors with CMYK and hex it was easier to work in hex. I can find the RGB
equivalents, though, and update colors throughout the code. Or use a
converter, a VS Code addon will likely do that I imagine?*

> Reviews.css hex decimal codes missing characters in the classes
> .review-header, .reviews-div
> and .product-search-button:hover has extra hex decimal code characters
>
*If they are missing characters it's likely fff for white, as you don't
need full 6 characters in those instances (but I try to put them in
anyways, I likely just overlooked adding the extra f's. Additional
characters adjust opacity and are functioning correctly. I'll double check
them all.*

> Sell.css .user-signup-redirect:hover hex decimal code has extra characters
>
> Notes on frontend code, stylistically speaking it's well written. Few
> things I would like to express to you. When someone uses single quotes
> originally you can keep the single quotes no need to change them to doubles
> since JavaScript recognizes both for strings. if it was C languages or
> Python that would be a good catch. Sticking with the same properties is
> also something to consider, I use rgb because its easy to write for me, hex
> decimal codes I have to look up every time. there is sites that convert hex
> decimals to rgb. making sure everything is similar is always good when
> collaborating because it keeps everything nice and organized.
>
*I didn't replace any of your quotes on purpose. If quotes were replaced it
was just an accident. I went back and forth with each page completely
changing the code, I didn't intentionally go through and adjust quotes.
Would you like me to find and replace to change double quotes to single?*

> I would like you to fix whatever happened to the node_modules in the
> server folder, Why did you rewrite these? Its very bad practice to rewrite
> someone else's code without their permission especially node modules or npm
> packages that have been published. I specifically asked in the beginning
> that you stayed out of this folder, and should only go into it to install
> the modules, so you could run the files. Whatever happened in this folder
> it wants to delete the entire package-lock.json which would destroy the
> backend entirely. this has to be fixed before I can merge this branch
>
*If the node_modules were rewritten it was from some command I ran when
trying to get the site to work. I have no experience with node modules and
never touched anything in that folder. I wouldn't have rewritten any of
your code, outside of styles, without telling you. When I did rewrite
something (the TextField to input) I was sure to ask you about it, because
I wanted to be very careful not to mess up anything you'd done. I have no
experience outside of frontend styles, and that is all I intentionally
adjusted. Is there anything you can glean from what is changed in that
folder to have an idea of what I may have done so I can be sure to avoid
doing that again? I assure you, anything that happened there was
unintentional and I never went into that folder to edit anything.*

> other than the stylistic differences, the code is well written and
> readable.
>
*Thank you! I appreciate the feedback and patience with anything I get
wrong from inexperience or ignorance to proper process.*

> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NekoCarreraDesigns/eformative/pull/2#pullrequestreview-1437228141>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS5W4C5OAPS73TCI6HHRTWLXHOWMRANCNFSM6AAAAAAYFNMRHI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

*Mark Hager*
417.499.3710
www.HagerCreative.com
